great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2829,8 +2835,8 @@ def _perform_substitutions_on_datasource_config(
     def _instantiate_datasource_from_config_and_update_project_config(
         self,
         config: DatasourceConfig,
+        save_changes: bool,  # Specifically deviate from other uses of this flag due to usage in `test_yaml_config`
```

Comment:
Question - should `test_yaml_config` impact state at all (context and/or disk)?

I think not but it looks like when this method is used, we're updating the config of the context in question.